src/pipeline_edsl/pipeline_library.mli

```diff
@@ -4,15 +4,25 @@ module Input : sig
 
   type t =
     | Fastq of fastq
+    | Bam of bam
   and fastq_fragment = (string option * fastq_data)
   and fastq = {
-    sample_name : string;
+    fastq_sample_name : string;
     files : fastq_fragment list;
   }
+  and bam = {
+    bam_sample_name: string;
+    path: string;
+    how: [ `PE | `SE ];
+    sorting: sorting option;
+    reference_build: string;
+  }
   and fastq_data =
     | PE of string * string
     | SE of string
     | Of_bam of [ `PE | `SE ] * [ `Coordinate | `Read_name ] option * string * string
```

Comment:
Also this looks like it should be factored, at least using the `how` and `sorting` types :)